client/wfprev-war/src/main/angular/src/app/components/edit-project/fiscal-map/fiscal-map.component.ts

```diff
@@ -0,0 +1,318 @@
+      this.projectService.getProjectFiscalsByProjectGuid(this.projectGuid).subscribe((data) => {
+        this.projectFiscals = (data._embedded?.projectFiscals || []).sort(
+          (a: { fiscalYear: number }, b: { fiscalYear: number }) => a.fiscalYear - b.fiscalYear
+        );
+  
+        const activityRequests = this.projectFiscals.map(fiscal =>
+          this.projectService.getFiscalActivities(this.projectGuid, fiscal.projectPlanFiscalGuid).pipe(
+            map((response: any) => {
+              const activities = response?._embedded?.activities || [];
+              return activities.map((activity: any) => ({
+                ...activity,
+                fiscalYear: fiscal.fiscalYear,
+                projectPlanFiscalGuid: fiscal.projectPlanFiscalGuid
+              }));
+            })
+          )
+        );
+  
+        forkJoin(activityRequests).subscribe((allActivityArrays) => {
+          const allActivities = allActivityArrays.flat();
+  
+          if (allActivities.length === 0) {
+            // no activites at all.
+            this.getProjectCoordinates(); 
+            return;
+          }
+          const boundaryRequests = allActivities.map(activity =>
+            this.projectService
+              .getActivityBoundaries(this.projectGuid, activity.projectPlanFiscalGuid, activity.activityGuid)
+              .pipe(
+                map(boundary => boundary ? ({
+                  activityGuid: activity.activityGuid,
+                  fiscalYear: activity.fiscalYear,
+                  boundary: boundary?._embedded?.activityBoundary
+                }) : null),
+              )
+          );
+  
+          forkJoin(boundaryRequests).subscribe((allResults) => {
+            this.allActivityBoundaries = allResults.filter(r => 
+              r !== null && r.boundary && Object.keys(r.boundary).length > 0
+            );
+            const hasActivityPolygons = this.allActivityBoundaries.length > 0;
+            const hasProjectPolygons = this.projectBoundary?.length > 0;
+            
+            if (hasActivityPolygons && this.map) {
+              this.plotBoundariesOnMap(this.allActivityBoundaries);
+            }
+            
+            //  Only show pin if NO polygons exist at all
+            if (!hasActivityPolygons && !hasProjectPolygons) {
+              this.getProjectCoordinates();
+            }
+          });
+        });
+      });
+    }
+  }
+  
+  plotBoundariesOnMap(boundaries: any[]): void {
+    const currentFiscalPolygons: L.Layer[] = [];
+  
+    boundaries.forEach(boundaryEntry => {
+      const fiscalYear = boundaryEntry.fiscalYear;
+      let color = '';
+  
+      if (fiscalYear < this.currentFiscalYear) {
+        color = this.fiscalColorMap.past;
+      } else if (fiscalYear === this.currentFiscalYear) {
+        color = this.fiscalColorMap.present;
+      } else {
+        color = this.fiscalColorMap.future;
+      }
+  
+      boundaryEntry.boundary.forEach((item: any) => {
```

Comment:
It is almost always better to use `for(let thing of things)` vs an `array.forEach`. It's generally clearer to read and debug, and also performs better

Reply:
Sure, changed to use for loop